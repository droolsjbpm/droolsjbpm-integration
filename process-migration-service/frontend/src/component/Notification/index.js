import React from "react";
import PropTypes from "prop-types";
import {
  ToastNotificationList,
  TimedToastNotification
} from "patternfly-react";

export default class Notification extends React.Component {
  constructor(props) {
    super(props);
  }

  render() {
    return (
      <ToastNotificationList>
        <TimedToastNotification
          type={this.props.type}
          persistent={false}
          onDismiss={this.props.onDismiss}
        >
          {this.props.message}
        </TimedToastNotification>
      </ToastNotificationList>
    );
  }
}

Notification.propTypes = {
  type: PropTypes.string.isRequired,
  message: PropTypes.string.isRequired,
  onDismiss: PropTypes.func.isRequired
};
