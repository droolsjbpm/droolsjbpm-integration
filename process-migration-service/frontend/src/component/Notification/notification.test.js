import React from "react";
import renderer from "react-test-renderer";
import Notification from "./";

test("Notification tests", () => {
  const mock = jest.fn();
  const tree = renderer
    .create(
      <Notification type="error" message="Test message" onDismiss={mock} />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
