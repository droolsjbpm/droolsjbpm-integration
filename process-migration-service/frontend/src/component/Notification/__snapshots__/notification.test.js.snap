// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Notification tests 1`] = `
<div
  className="toast-notifications-list-pf"
  onFocus={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onMouseOver={[Function]}
>
  <div
    className="alert toast-pf alert-danger alert-dismissable"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <button
      aria-hidden="true"
      className="close close-default"
      disabled={false}
      onClick={[MockFunction]}
      type="button"
    >
      <span
        aria-hidden="true"
        className="pficon pficon-close"
      />
    </button>
    <span
      aria-hidden="true"
      className="pficon pficon-error-circle-o"
    />
    Test message
  </div>
</div>
`;
